src/concepts/segments.md

```diff
@@ -1,75 +1,106 @@
 as opposed to more stable traits (e.g. country), 
-we should evaluate each user's segment eligibility 
+we should evaluate each user's user state eligibility 
 using data collected _before_ the time period in which we want to study their actions. 
 Else we run the risk of making trivial discoveries 
 like "heavy users use the product heavily" instead of more meaningful ones 
 like "heavy users go on to continue using the product heavily".
 
-So, when writing queries to compute segments directly from their definition, 
-be sure to compute users' segments using only 
+So, when writing queries to compute user states directly from their definition, 
+be sure to compute users' user states using only 
 data collected before the time period you're analyzing their behavior.
 
-Segments are found as columns in the `clients_last_seen` dataset: the segment listed for a client on a `submission_date` is valid for that `submission_date` because it is computed only using behavioral data collected _before_ the `submission_date`.
+User states are found as columns in the `clients_last_seen` dataset: the user state listed for a client on a `submission_date` is valid for that `submission_date` because it is computed only using behavioral data collected _before_ the `submission_date`.
 
 ### WAU and MAU
 
-Users can change segments part way through a week or a month. 
-At present, 
-we have not settled the questions of whether and how to split MAU by segment,
-and whether and how to measure MAU for each segment.
-So stick to DAU for now.
+Users can move in or out of specific user states part way through a week or a month.
+This poses a conundrum if we want to plot the WAU or MAU for a user state.
+
+Our convention is to _count the number of distinct users who were active in user state in the period_.
+So if a user was active as a regular user v3 on e.g. the second day of a 28-day window, then they will contribute to "regular user v3 MAU" regardless of whether they lost their "regular user v3" status at any point in the 28-day window.
+
+Since many user states use the full extent of the `*_bits` column wizardry in `clients_last_seen`, you'll have to query WAU or MAU the old fashioned way:
+
+```lang=sql
+WITH dates AS (
+    SELECT *
+    FROM UNNEST(GENERATE_DATE_ARRAY('2020-05-01', '2020-07-01')) as d
+) SELECT
+    dates.d AS submission_date,
+    COUNT(DISTINCT client_id) * 100 AS regular_user_v3_mau,
+FROM dates
+INNER JOIN telemetry.clients_last_seen cls
+    ON cls.submission_date BETWEEN DATE_SUB(dates.d, INTERVAL 27 DAY) AND dates.d
+    AND cls.submission_date BETWEEN '2020-04-01' AND '2020-07-01'
+WHERE cls.sample_id = 42
+    AND cls.is_regular_user_v3
+    AND cls.days_since_seen = 0
+GROUP BY dates.d
+ORDER BY dates.d
+```
+
+Our convention has the potentially counterintuitive consequence that a user can count towards "regular user v3 MAU" and "not regular user v3 MAU" for the same 28-day window.
```

Comment:
This sentence is where I see the terminology really break down. The idea of MAU for "not regular user v3" is problematic because we are marking users as "active" for a specific definition of _not_ being active.

This quantity may well be a useful one to calculate and observe, but I'd like to drop the MAU/WAU terminology at this point. This could be something like "monthly unique users in state"?

Reply:
> we are marking users as "active" for a specific definition of not being active

It's marking users as "active as not a regular user" (i.e. their browser submitted a ping on a day they didn't qualify to be a regular user), similar to how we'd mark users as "active not in the US" vs "active in the US". Do I need a better segment name than "not regular user" for "not regular user v3 MAU", or is it the concept itself the issue?

"monthly unique users in state" loses the implication that the users were active while being in that state - and it's important to keep that implication, otherwise for example regular users can never churn without appearing to transition through other states including "new or resurrected"(!)

Reply:
I think that @jklukas makes a good suggestion about not referring to this quantity as MAU.  You can certainly use it, but having another name makes its strange behaviours less potentially-confusing.

I would still support the end-of-window definition that you describe below as the MAU definition.  Of your two arguments against it in the text below:

1) I'm sure that being a end-of-window regular user MAU is also predictive of retention and, in any case, predicting retention isn't the only reason to care about regular users
2) Being a new profile is something that happens on a discrete day.  Using MAU of a segment is not really a reasonable way to try to count discrete events.

Reply:
Further out-of-band communication with @jklukas resulted in the "MAU(sent a ping as a regular user v3)" language that's in the current version of this PR. The current version of the text tries to establish "regular user v3 MAU" as a shorthand for the verbose "MAU(sent a ping as a regular user v3)"

> I'm sure that being a end-of-window regular user MAU is also predictive of retention and, in any case, predicting retention isn't the only reason to care about regular users

It wasn't designed to be used in this way; if we want a quantity that works optimally for this then we should specifically design a quantity for this purpose, and it will likely be a different quantity.

Reply:
Is it necessary that we apply user state onto WAU and MAU? 

DAU can be split into 3 exclusive user states. For example
         `DAU = Regular user v3 DAU +  New or resurrected v3 DAU + (Other v3 DAU)`, 
where 
        `other` = `(not is_regular_user_v3) and (not is_new_or_resurrected_v3)`

We can associate change in DAU with change in user state DAU; but not for WAU or MAU. As a user can change user state over a period of time, following equality won’t establish
          `MAU = Regular user v3 MAU +  New or resurrected v3 MAU + (Other v3 MAU)`, 

The left side will always be less than the right side, and the wider the window the larger the difference will be (as the higher the chance that a user is going to change user state); as seen in [here](https://sql.telemetry.mozilla.org/queries/73184/source#183289) and [here](https://sql.telemetry.mozilla.org/queries/73185/source#183288)

Reply:
Talked to Felix and he has clarified my above concern on the breakdown of WAU and MAU per user states.